modules/audio/inference/siren_detection.cc

```diff
@@ -0,0 +1,86 @@
+/******************************************************************************
+ * Copyright 2020 The Apollo Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *****************************************************************************/
+
+#include "modules/audio/inference/siren_detection.h"
+
+#include <vector>
+#include <utility>
+
+#include <omp.h>
+
+#include "cyber/common/log.h"
+#include "modules/audio/common/audio_gflags.h"
+
+namespace apollo {
+namespace audio {
+
+SirenDetection::SirenDetection() : device_(torch::kCPU) {
+  LoadModel();
+}
+
+bool SirenDetection::Evaluate(const std::vector<std::vector<float>>& signals) {
+  // Sanity checks.
+  omp_set_num_threads(1);
+  if (signals.size() == 0) {
+    AERROR << "Got no channel in signals!";
+    return false;
+  }
+  if (signals[0].size() == 0) {
+    AERROR << "Got no signal in channel 0!";
+    return false;
+  }
+
+  torch::Tensor audio_tensor = torch::empty(6 * 1 * 30075);
+  float* data = audio_tensor.data_ptr<float>();
+
+  for (const auto& channel : signals) {
+    for (const auto& i : channel) {
+      *data++ = i;
+    }
+  }
+
+  torch::Tensor torch_input = torch::from_blob(data, {6, 1, 30075});
+  std::vector<torch::jit::IValue> torch_inputs;
+  torch_inputs.push_back(std::move(torch_input.to(device_)));
```

Comment:
AIUI, the temporary value returned by `torch_input.to(device_)` is  `rvalue` reference. No need to call `std::move` explicitly here.